### PR TITLE
Fix issue when retrieving scim groups through IDN_SCIM_GROUP table

### PR DESCRIPTION
**Pull request by GANGANI**
Resolves: [https://github.com/wso2/product-is/issues/7412](https://github.com/wso2/product-is/issues/7412)

### CODE REVIEW

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java [hunk 1]
```diff
@@ -79,6 +79,40 @@
         return groups;
     }
 
+    /**
+     * Lists the groups that belong to a tenant by its tenant id that are created from SCIM
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java]: Shall we add a `.` at the end of the line?

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java [hunk 2]
```diff
@@ -79,6 +79,40 @@
...
+     *
+     * @param tenantId tenant Id
+     * @return The set of groups that were created from SCIM
+     * @throws IdentitySCIMException Error when reading from persistence store.
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java]: Shall we add **If an error occurred while ...**

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java [hunk 3]
```diff
@@ -79,6 +79,40 @@
...
+     */
+    public Set<String> listSCIMGroups(int tenantId) throws IdentitySCIMException {
+        Connection connection = IdentityDatabaseUtil.getDBConnection();
+        PreparedStatement prepStmt = null;
+        ResultSet resultSet = null;
+        Set<String> groups = new HashSet<>();
+
+        try {
+            //retrieve groups from the DB
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java]: `.` at the end of the comment and start the comment with an uppercase letter.

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java [hunk 4]
```diff
@@ -79,6 +79,40 @@
...
     }
 
+    /**
+     * Lists the groups that belong to a tenant by its tenant id that are created from SCIM
+     *
+     * @param tenantId tenant Id
+     * @return The set of groups that were created from SCIM
+     * @throws IdentitySCIMException Error when reading from persistence store.
+     */
+    public Set<String> listSCIMGroups(int tenantId) throws IdentitySCIMException {
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java]: Formatting issue.

Comment (GANGANI) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java]: Fixed with https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/252/commits/98c7dd4b02908f8feb007d39d8f7fca4a7abae6d

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java (hunk 3 above)
Comment (GANGANI) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java]: Fixed with https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/252/commits/98c7dd4b02908f8feb007d39d8f7fca4a7abae6d

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java (hunk 2 above)
Comment (GANGANI) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java]: Fixed with https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/252/commits/98c7dd4b02908f8feb007d39d8f7fca4a7abae6d

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java (hunk 1 above)
Comment (GANGANI) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java]: Fixed with https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/252/commits/98c7dd4b02908f8feb007d39d8f7fca4a7abae6d

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java [hunk 5]
```diff
@@ -79,6 +83,41 @@
...
+    /**
+     * Lists the groups that belong to a tenant by its tenant id that are created from SCIM.
+     *
+     * @param tenantId tenant Id
+     * @return The set of groups that were created from SCIM
+     * @throws IdentitySCIMException If an error occurred while reading from persistence store.
+     */
+    public Set<String> listSCIMGroups(int tenantId) throws IdentitySCIMException {
+
+        Connection connection = IdentityDatabaseUtil.getDBConnection();
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java]: use try with resource for connections, prepared statements and result sets.

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java [hunk 6]
```diff
@@ -79,6 +83,41 @@
...
+            while (resultSet.next()) {
+                String group = resultSet.getString(1);
+                if (StringUtils.isNotEmpty(group)) {
+                    group = SCIMCommonUtils.getPrimaryFreeGroupName(group);
+                    groups.add(group);
+                }
+            }
+        } catch (SQLException e) {
+            throw new IdentitySCIMException("Error when reading the SCIM Group information from persistence store.", e);
+        } finally {
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java]: finally block can be removed when the try with resource is used.

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java (hunk 5 above)
Comment (GANGANI) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java]: Added with https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/252/commits/734e0608b7a8b0b0f7e299af86bda0d092285945

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java (hunk 6 above)
Comment (GANGANI) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java]: Removed with https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/252/commits/734e0608b7a8b0b0f7e299af86bda0d092285945

**Pull request merged**